fix(php): Correct uint32 type mapping to use UInt32 instead of Int32

TomA-R opened the pull request:
This fixes the PHP extension's mapping for `uint32`, which was incorrectly using `Int32`. I've also added a couple of small tests to cover edge cases in `RepeatedField` and `MapField`

honglooker commented on php/ext/google/protobuf/tests/php_uint32_mapfield_coercion.phpt:
```diff
@@ -0,0 +1,43 @@
+--TEST--
```
Mind adding a blurb on copyright? In cpp code, it's generally of the form:
```
// Protocol Buffers - Google's data interchange format
// Copyright 2025 Google LLC.  All rights reserved.
//
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file or at
// https://developers.google.com/open-source/licenses/bsd
```

Please convert this to the equivalent in php tests - thank you

TomA-R replied:
I've removed this file in favour of test additions to php/tests/GeneratedClassTest.php

honglooker commented on php/ext/google/protobuf/tests/php_uint32_repeatedfield_coercion.phpt:
```diff
@@ -0,0 +1,42 @@
+--TEST--
```
same here

TomA-R replied:
I've removed this file in favour of test additions to php/tests/GeneratedClassTest.php

honglooker reviewed (commented):
Thanks for your patch! Added commentary below.

honglooker commented:
Alternatively, it may also make sense to augment php/tests/GeneratedClassTest.php instead.

TomA-R commented:
> Alternatively, it may also make sense to augment php/tests/GeneratedClassTest.php instead.

Thank you for your review, @honglooker! I've added the tests to php/tests/GeneratedClassTest.php and removed the original PHPT tests. I hope that's what you expected, let me know if you'd like me to make any other changes

honglooker commented on php/tests/GeneratedClassTest.php:
```diff
@@ -269,6 +269,43 @@ public function testUint32Field()
+        $m = new TestMessage();
+        $map = $m->getMapUint32Uint32();
+
+        $map[1] = 0;
```
why not map[0] ?

honglooker reviewed (commented):
Thanks, some q's below
